File: app/src/main/java/com/example/myittaroostockinventorymanger/stock_name_fragment/ConfirmDialogFragment.java
package com.example.myittaroostockinventorymanger.stock_name_fragment;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.lifecycle.ViewModelProvider;

import com.example.myittaroostockinventorymanger.R;
import com.example.myittaroostockinventorymanger.local.Stock;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ConfirmDialogFragment extends DialogFragment {

    @BindView(R.id.btn_yes)
    Button btnYes;
    @BindView(R.id.btn_no)
    Button btnNo;

    private AlertDialog alertDialog;
    private Context context;
    private Stock stock;

    private ConfirmDialogFragment() {

    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        LayoutInflater layoutInflater = getLayoutInflater();
        View view = layoutInflater.inflate(R.layout.confirm_dialog_fragment, null, false);
        ButterKnife.bind(this, view);

        alertDialog = new AlertDialog.Builder(getContext())
                .setView(view)
                .create();

        alertDialog.getWindow()
                .setBackgroundDrawableResource(R.drawable.rounded_rectangle_white);

        Bundle bundle = getArguments();

        if (bundle != null) {
            stock = bundle.getParcelable(StockNameFragment.EXTRA_DELETE);
        }

        return alertDialog;

    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        ConfirmDialogFragmentViewModel confirmDialogFragmentViewModel = new ViewModelProvider(requireActivity())
                .get(ConfirmDialogFragmentViewModel.class);

        btnYes.setOnClickListener(v -> {
            confirmDialogFragmentViewModel.deleteStock(stock);
            alertDialog.cancel();
        });

        btnNo.setOnClickListener(v -> {
            alertDialog.cancel();
        });

        confirmDialogFragmentViewModel.getMessage()
                .observe(getParentFragment().getViewLifecycleOwner(), m -> {

                    String message = m.getContentIfNotHandle();

                    if (message != null) {
                        Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
                    }
                });

        return super.onCreateView(inflater, container, savedInstanceState);
    }

    public static ConfirmDialogFragment getNewInstance(String key, Stock stock) {
        ConfirmDialogFragment confirmDialogFragment = new ConfirmDialogFragment();
        Bundle bundle = new Bundle();
        bundle.putParcelable(key, stock);
        confirmDialogFragment.setArguments(bundle);
        return confirmDialogFragment;
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        this.context = context;
    }
}

File: app/src/main/java/com/example/myittaroostockinventorymanger/ui/batch/ConfirmDialogFragment.kt
package com.example.myittaroostockinventorymanger.ui.batch

import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.viewModels
import androidx.navigation.fragment.navArgs
import com.example.myittaroostockinventorymanger.Application
import com.example.myittaroostockinventorymanger.R
import com.example.myittaroostockinventorymanger.databinding.ConfirmDialogFragmentBinding

class ConfirmDialogFragment : DialogFragment() {

    private lateinit var context: Context
    private lateinit var alertDialog: AlertDialog
    private lateinit var binding: ConfirmDialogFragmentBinding
    private var selectedIdList: List<Long> = arrayListOf()
    private val args: ConfirmDialogFragmentArgs by navArgs()
    private val viewModel: ConfirmDialogViewModel by viewModels()

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        binding = ConfirmDialogFragmentBinding.inflate(layoutInflater, null, false)
        val view = binding.root
        context = Application.getContext()

        selectedIdList = args.selectedIdList.toList()
        Log.d("myTag", "onCreateDialog: $selectedIdList")

        alertDialog = AlertDialog.Builder(getContext())
            .setView(view)
            .create()

        alertDialog.window
            ?.setBackgroundDrawableResource(R.drawable.rounded_rectangle_white)

        return alertDialog
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        viewModel.setTitle(selectedIdList.size)

        viewModel.getTitle()
            .observe(this) {
                binding.txtTitle.text = it
            }

        binding.btnYes.setOnClickListener {
            viewModel.onClickYes(selectedIdList)
            alertDialog.cancel()
        }

        binding.btnNo.setOnClickListener {
            alertDialog.cancel()
        }

        viewModel.getMessage()
            .observe(this) {
                val notify = it.contentIfNotHandle
                if (notify != null) {
                    Toast.makeText(context, notify, Toast.LENGTH_SHORT).show()
                }
            }
        return super.onCreateView(inflater, container, savedInstanceState)
    }

}